lib/mkmf.rb

```diff
@@ -1957,11 +1957,34 @@ def fmt.%(x)
         pkgconfig = nil
       end
       if pkgconfig
+        has_ms_win_syntax = false
+        if $mswin
+          has_ms_win_syntax = xpopen([pkgconfig, "--help"]).read.include?('msvc-syntax')
+          if has_ms_win_syntax
+            args << "--msvc-syntax"
+          else
+            Logging.message("WARNING: #{pkgconfig} does not support the --msvc-syntax. Try using a recent pkgconf instead")
+          end
+        end
         get = proc {|opts|
           opts = Array(opts).map { |o| "--#{o}" }
           opts = xpopen([*envs, pkgconfig, *opts, *args], err:[:child, :out], &:read)
           Logging.open {puts opts.each_line.map{|s|"=> #{s.inspect}"}}
-          opts.strip if $?.success?
+          if $?.success?
+            opts = opts.strip
+            if $mswin and not has_ms_win_syntax
+              opts = Shellwords.shellwords(opts).map { |s|
+                if s.start_with?('-l')
+                  "#{s[2..]}.lib"
+                elsif s.start_with?('-L')
+                  "/libpath:#{s[2..]}"
+                else
+                  s
+                end
+              }.quote.join(" ")
+            end
```

Comment:
Maybe msvc-syntax is not needed?

```suggestion
            libarg, libpath = LIBARG, LIBPATHFLAG.strip
            opts = opts.shellsplit.map { |s|
              if s.start_with?('-l')
                libarg % s[2..]
              elsif s.start_with?('-L')
                libpath % s[2..]
              else
                s
              end
            }.quote.join(" ")
```

Reply:
Yeah I don't know. I kinda like the idea of defering to the pkgconf since that's the tool's job. But maybe there isn't much to do to switch the syntax from Nix to MSVC.

Up to you.

The MSVC renderer from pkgconf is here, doesn't appear to do a lot of complicated things: https://github.com/pkgconf/pkgconf/blob/b6e04e2d47b4ca093632b9efde6303a137ea0634/cli/renderer-msvc.c#L133